frontend/src/context/Toast/ToastContext.tsx

```diff
@@ -0,0 +1,204 @@
+  useCallback,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { createPortal } from 'react-dom';
+import { styled } from 'styled-components';
+
+import Toast from '@/components/@common/Toast/Toast';
+import { Type } from '@/constants/toast';
+import { Toast as ToastInterface, ToastContent, ToastId, ToastOption } from '@/types/toast/client';
+
+interface ToastContext {
+  toast: {
+    (content: ToastContent, option?: ToastOption): ToastInterface;
+    success: (content: ToastContent, option?: ToastOption) => ToastInterface;
+    info: (content: ToastContent, option?: ToastOption) => ToastInterface;
+    warning: (content: ToastContent, option?: ToastOption) => ToastInterface;
+  };
+  currentToast: ToastInterface[];
+}
+
+const tempToast = (content: ToastContent, option?: ToastOption): ToastInterface => ({
+  content: '',
+  option: {
+    toastId: -1,
+    type: Type.INFO,
+  },
+});
+
+tempToast.success = (content: ToastContent, option?: ToastOption): ToastInterface => ({
+  content: '',
+  option: {
+    toastId: -1,
+    type: Type.SUCCESS,
+  },
+});
+
+tempToast.info = (content: ToastContent, option?: ToastOption): ToastInterface => ({
+  content: '',
+  option: {
+    toastId: -1,
+    type: Type.INFO,
+  },
+});
+
+tempToast.warning = (content: ToastContent, option?: ToastOption): ToastInterface => ({
+  content: '',
+  option: {
+    toastId: -1,
+    type: Type.WARNING,
+  },
+});
+
+const ToastContext = createContext<ToastContext>({
+  toast: tempToast,
+  currentToast: [],
+});
+
+ToastContext.displayName = 'Toast';
+
+export const useToast = () => useContext(ToastContext);
+
+interface ToastProviderProps {}
+
+const Portal = ({ children }: PropsWithChildren) => createPortal(children, document.body);
+
+const ToastProvider = (props: PropsWithChildren<ToastProviderProps>) => {
+  const { children } = props;
+
+  const [toastStates, setToastStates] = useState<ToastInterface[]>([]);
+  const currentToast = useRef<ToastInterface[]>([]);
+
+  const toastId = useRef<ToastId>(0);
+
+  const timers = useRef<Map<ToastId, NodeJS.Timeout>>(new Map<ToastId, NodeJS.Timeout>());
+
+  const deleteToastId = useRef<ToastId>(0);
+
+  const appendToast = useCallback((toast: ToastInterface) => {
+    currentToast.current = [...currentToast.current, toast];
+
+    toastId.current += 1;
+
+    const timer = setTimeout(() => {
+      deleteToast(toast.option.toastId);
+    }, 4500);
+
+    timers.current.set(toast.option.toastId, timer);
+
+    setToastStates(currentToast.current);
+  }, []);
+
+  const deleteToast = (toastId: ToastId) => {
+    const targetToastIndex = currentToast.current.findIndex(
+      toast => toast.option.toastId === toastId,
+    );
+    if (targetToastIndex !== -1) {
+      const copyArr = currentToast.current.slice();
+      copyArr.splice(targetToastIndex, 1);
+
+      currentToast.current = copyArr;
+
+      setToastStates(currentToast.current);
+
+      timers.current.delete(toastId);
+    }
```

Comment:
```suggestion
        currentToast.current =  currentToast.current.filter((toast)=> toast.option.toastId !== toasId)
    
    setToastStates(currentToast.current);

      timers.current.delete(toastId);
```

`filter`를 사용하는건 어떨까요!?

Reply:
굳굳 반영했습니다!